source/extensions/matching/input_matchers/ip/matcher.cc

```diff
@@ -1,30 +1,43 @@
-#include "extensions/matching/input_matchers/ip/matcher.h"
+#include "source/extensions/matching/input_matchers/ip/matcher.h"
 
-#include "common/network/utility.h"
+#include "source/common/network/utility.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace Matching {
 namespace InputMatchers {
 namespace IP {
 
-Matcher::Matcher(std::vector<Network::Address::CidrRange>&& ranges)
-    // We could put "false" instead of "true". What matters is that the IP
-    // belongs to the trie. We could further optimize the storage of LcTrie in
-    // this case by implementing an LcTrie<void> specialization that doesn't
-    // store any associated data.
-    : trie_({{true, std::move(ranges)}}) {}
+Matcher::Matcher(std::vector<Network::Address::CidrRange>&& ranges, absl::string_view stat_prefix,
+                 Stats::Scope& stat_scope)
+    : // We could put "false" instead of "true". What matters is that the IP
+      // belongs to the trie. We could further optimize the storage of LcTrie in
+      // this case by implementing an LcTrie<void> specialization that doesn't
+      // store any associated data.
+      trie_({{true, std::move(ranges)}}) {
+  if (!stat_prefix.empty()) {
+    stats_.emplace(generateStats(stat_prefix, stat_scope));
+  }
```

Comment:
I think its more common to emit stats by default for extensions, so maybe make stats_prefix required so that it's always set? Or is there a good reason to avoid emitting this stat?

Reply:
I've seen this in another extension, so I just made the same :) I have no other rational than this, I'm happy to make it mandatory!

Reply:
Done in https://github.com/envoyproxy/envoy/pull/16592/commits/7960aa3c22f33d667628bc98a0e885dcb183139e !